storagemarket/deal_commp.go

```diff
@@ -56,6 +65,13 @@ func (p *Provider) generatePieceCommitment(filepath string, pieceSize abi.Padded
 		var err error
 		pi, err = GenerateCommP(filepath)
 		if err != nil {
+			// Allow auto retry to cover cases where IO copy fails due to lack of space
+			if strings.Contains(err.Error(), "failed to write to CommP writer") {
```

Comment:
Instead of doing string matching on the error string, let's use a custom error.
See for example ErrDealHandlerNotFound
https://github.com/filecoin-project/boost/blob/015b65463ca4499bbbb726a1f1477473bd56d12c/gql/resolver.go#L178